Apps/WebClient/src/ClientApp/src/components/modal/addQuickLink.vue

```diff
@@ -0,0 +1,321 @@
+                name: "My Notes",
+                module: ClientModule.Note,
+                isEnabled: this.webClientConfig.modules[EntryType.Note],
+            },
+            {
+                name: "Special Authority",
+                module: ClientModule.MedicationRequest,
+                isEnabled:
+                    this.webClientConfig.modules[EntryType.MedicationRequest],
+            },
+        ];
+    }
+
+    private get enabledQuickLinkFilter(): QuickLinkFilter[] {
+        this.logger.debug(
+            `Enabled Quick Links - checkbox component key:  ${this.checkboxComponentKey}`
+        );
+        if (this.checkboxComponentKey != 0) {
+            return this.enabledQuickLinks;
+        } else {
+            this.logger.debug(
+                `Enabled Quick Links - quick links:  ${this.quickLinks}`
+            );
+            if (this.quickLinks != undefined) {
+                this.populateEnabledQuickLinks();
+                const existingQuickLinks = this.quickLinks;
+                return this.getAvailabeQuickLinks(
+                    existingQuickLinks,
+                    this.enabledQuickLinks
+                );
+            }
+            this.populateEnabledQuickLinks();
+            return this.enabledQuickLinks;
+        }
+    }
+
+    private get isAddButtonDisabled(): boolean {
+        return (
+            this.enabledQuickLinkFilter.length == 0 ||
+            (this.enabledQuickLinkFilter.length > 0 &&
+                this.selectedQuickLinks.length == 0)
+        );
+    }
+
+    private get isCheckBoxChecked(): boolean {
+        return this.enabledQuickLinkFilter.length == 0;
+    }
+
+    private addQuickLink(submittedQuickLinks: QuickLink[]): Promise<void> {
+        const quickLinks: QuickLink[] = (this.quickLinks ?? []).concat(
+            submittedQuickLinks
+        );
+        return this.updateQuickLinks({
+            hdid: this.user.hdid,
+            quickLinks: quickLinks,
+        });
+    }
+
+    private created() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    }
+
+    private getAvailabeQuickLinks(
+        existingQuickLinks: QuickLink[],
+        enabledQuickLinks: QuickLinkFilter[]
+    ): QuickLinkFilter[] {
+        existingQuickLinks.forEach((element) =>
+            this.logger.debug(
+                `Available Quick Links - Existing store quick links:  ${element}`
+            )
+        );
+
+        enabledQuickLinks.forEach((element) =>
+            this.logger.debug(
+                `Available Quick Links - Enabled quick links:  ${element}`
+            )
+        );
+
+        let result: QuickLinkFilter[] = [];
+        result = enabledQuickLinks.filter((enabledQL) => {
+            return !existingQuickLinks.find((existingQL) => {
+                return existingQL.name === enabledQL.name;
+            });
+        });
+        return result;
+    }
+
+    private getCheckedValue(quickLink: QuickLinkFilter): string {
```

Comment:
The checkboxes could be simplified so they are associated with just the module and the QuickLink could be populated from that later.

Reply:
Done.